packages/utilities/adaptive-ui/src/color/palette.ts

```diff
@@ -0,0 +1,231 @@
+import { clamp } from "@microsoft/fast-colors";
+import { Swatch } from "./swatch.js";
+import { binarySearch } from "./utilities/binary-search.js";
+import { directionByIsDark } from "./utilities/direction-by-is-dark.js";
+import { contrast, RelativeLuminance } from "./utilities/relative-luminance.js";
+
+/**
+ * Directional values for navigating {@link Swatch}es in {@link Palette}.
+ *
+ * @public
+ */
+export enum PaletteDirectionValue {
```

Comment:
Can we use a const object here instead of an enum? Enums don’t compress and don’t technically exist in JS, so a const object will fit better here.

Reply:
Yep. I couldn't find that const structure the other day, but I since saw the conversation about it and kept it for later. I'll update.

Reply:
For your convenience, here's a snippet from a guidance document that I'm contributing to:

Enums are not explicitly supported in JavaScript, so TypeScript developers sometimes use patterns like the following: 

```ts
enum MyEnum { 
  valueOne = 2, 
  valueTwo = 2 
} 
```

Or like the following: 

```ts
namespace MyEnum { 
  export const valueOne = 1; 
  export const valueTwo = 2; 
} 
```

Both patterns should be avoided when writing modern TypeScript. Instead leverage a frozen const object in combination with a same-named type to provide both compile-time and runtime safety while remaining native to JavaScript. Here’s an example: 

```ts
const MyEnum = Object.freeze({ 
  valueOne: 1, 
  valueTwo: 2 
} as const); 

export type MyEnum = typeof MyEnum[keyof typeof MyEnum]; 
```

Reply:
Cool, that's what I ended up with. I've noticed API Extractor has some issues with multiple exports named the same. I didn't run into it here, but I believe it was when referencing like `@link MyEnum` in other doc comments. I think there's supposed to be a way to index the right one, but I didn't find it. If we run into that again and figure it out we should add a note to the docs you included here.

Reply:
Turns out I did run into this. This ended up being to solution. Note the parentheses are required:
```
{@link (PaletteDirectionValue:variable)}
```